pilot/pkg/model/push_context.go

```diff
@@ -256,7 +256,7 @@ type processedDestRules struct {
 	// Map of dest rule host to the list of namespaces to which this destination rule has been exported to
 	exportTo map[host.Name]map[visibility.Instance]bool
 	// Map of dest rule host and the merged destination rules for that host
-	destRule map[host.Name]*config.Config
+	destRules map[host.Name][]*config.Config
```

Comment:
I am concerned about the searching performance

Reply:
Also the merging performance

Reply:
Yeah I don't see what the issue is. For all current cases, this will be a list of size 1.

If a user starts making use of this new feature, they could have a few items here. I cannot imagine this being a large number.

For example, we would have 10 items if _in the same namespace_, 10 distinct workload labels had customized DestinationRules for the same hostname. That seems extremely low probability, and even if it is, its a linear scale not n^2 so doesn't seem too bad?

Is there a concern I missed that makes this perform worse than this?